_posts/2019-05-20-mockiato-announcement.md

```diff
@@ -0,0 +1,170 @@
+
+To be able to specify the number of expected calls, the generated `expect_` methods return a `MethodCallBuilder`, which amongst other methods contains a `times` method.
+
+The `times` method accepts a usize value or a range. Here is the full list of supported values from the documentation:
+
+| Description           | Type               | Example |
+| --------------------- | ------------------ | ------- |
+| Exact amount of times | `u64`              | `3`     |
+| Any amount of times   | `RangeFull`        | `..`    |
+| At least              | `RangeFrom`        | `3..`   |
+| At most (exclusive)   | `RangeTo`          | `..3`   |
+| At most (inclusive)   | `RangeToInclusive` | `..3`   |
+| Between (exclusive)   | `Range`            | `3..4`  |
+| Between (inclusive)   | `RangeInclusive`   | `3..=4` |
+
+```rust
+#[cfg(test)]
+use mockiato::mockable;
+
+#[cfg_attr(test, mockable)]
+trait Greeter {
+    fn greet(&self, name: &str);
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn greet_the_world() {
+        let mut greeter = GreeterMock::new();
+
+        greeter
+            .expect_greet(|arg| arg.partial_eq("world"))
+            .times(1..2);
+        
+        greeter.greet("world");
+    }
+}
+```
+
+## Specifying return values
+The `MethodCallBuilder` also contains a `returns` method.
+
+This `returns` method allow you to specify a return value, which is returned when the corresponding mock method is invoked.
+
+```rust
+#[cfg(test)]
+use mockiato::mockable;
+
+#[cfg_attr(test, mockable)]
+trait Greeter {
+    fn greet(&self, name: &str) -> String;
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn greet_the_world() {
+        let mut greeter = GreeterMock::new();
+
+        greeter
+            .expect_greet(|arg| arg.partial_eq("world"))
+            .returns(String::from("Hello world"));
+
+        assert_eq!("Hello world", greeter.greet("world"));
+    }
+}
+```
+
+### Specifying different return values for the same call
+For each method in the `Greeter` trait a corresponding `expect_<method_name>_calls_in_order` method is generated.
+
+Ordered expectations can be used together with [`times`](#expecting-a-method-call-a-certain-amount-of-times).
+
+```rust
+#[cfg(test)]
+use mockiato::mockable;
+
+#[cfg_attr(test, mockable)]
+trait Greeter {
+    fn greet(&self, name: &str) -> String;
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn greet_the_world() {
+        let mut greeter = GreeterMock::new();
+
+        greeter
+            .expect_greet(|arg| arg.partial_eq("world"))
+            .returns(String::from("Hello world"));
+
+        greeter
+            .expect_greet(|arg| arg.partial_eq("world"))
+            .returns(String::from("Hi world"));
+
+        greeter.expect_greet_calls_in_order();
+
+        assert_eq!("Hello world", greeter.greet("world"));
+        assert_eq!("Hi world", greeter.greet("world"));
+    }
+}
+```
+
+## Limitations
```

Comment:
I feel like these subsections should simply be a single table

Reply:
17bcf2e